pkg/data/managers/base.go

```diff
@@ -79,35 +81,56 @@ func (m *baseManager) GetPageInfo(ctx context.Context, table string, page parame
 	span.SetTag("pageInfo.curent", pageInfo.Current)
 	span.SetTag("pageInfo.itemsPerPage", pageInfo.ItemsPerPage)
 
-	builder := m.GetQueryBuilder()
-
-	err = builder.
-		Select("COUNT(*)").
-		From(table).
-		Where(scope).
-		QueryRowContext(ctx).
-		Scan(&pageInfo.UnfilteredItemCount)
-	if err != nil {
-		return pageInfo, err
+	filterSQL := "1 = 1"
+	filterArgs := []interface{}{}
+	if filter != nil {
+		filterSQL, filterArgs, err = filter.ToSql()
+		if err != nil {
+			return pageInfo, err
+		}
 	}
 
-	if filter != nil {
-		err = builder.
-			Select("COUNT(*)").
-			From(table).
-			Where(scope).
-			Where(filter).
-			QueryRowContext(ctx).
-			Scan(&pageInfo.ItemCount)
+	scopeSQL := "1 = 1"
+	scopeArgs := []interface{}{}
+	if scope != nil {
+		scopeSQL, scopeArgs, err = scope.ToSql()
 		if err != nil {
 			return pageInfo, err
 		}
-	} else {
-		pageInfo.ItemCount = pageInfo.UnfilteredItemCount
+	}
+
+	query := replaceQuestionMarks(fmt.Sprintf(`
+SELECT COUNT(*) AS unfilteredItemCount, SUM(hit) AS itemCount FROM (
+  SELECT
+    *,
+    (CASE WHEN (%v) THEN 1 ELSE 0 END) AS hit
+  FROM %v
+  WHERE %v
```

Comment:
I think I've asked it many times already, why are you using `%v` if you expect a string? `%v` is for outputting debug values. Every format where you expect an argument to be a string or stringer must be `%s`.

Reply:
`%+v` is for debug values, but you are right, I'm always expecting strings here so the reflect usage shouldn't be needed here and can be optimized away.

Reply:
done :)

Reply:
it's about safety, not about optimization. If the argument is a string the format must be a string, otherwise you can get weird errors, we already had it a few times when a value was a pointer of nil.